rat/ast.py

```diff
@@ -578,6 +679,29 @@ def __str__(self):
         return f"InverseLogit({self.subexpr})"
 
 
+@dataclass
+class IfElse(Expr):
+    condition: Expr
+    true_expr: Expr
+    false_expr: Expr
+
+    def __iter__(self):
+        yield self
+        for expr in itertools.chain(self.condition, self.true_expr, self.false_expr):
+            yield expr
+
+    def __post_init__(self):
+        signatures = {
+            (types.BooleanType, types.IntegerType, types.IntegerType): types.IntegerType,
+            (types.BooleanType, types.RealType, types.RealType): types.RealType,
+            (types.BooleanType, types.NumericType, types.NumericType): types.NumericType,
```

Comment:
What's the difference in Integer/Real/Numeric?

Reply:
Numeric is Integer or Real. It's for those mixed operations so that you don't have to type out 4 type signatures.